### PR TITLE
Fix MVA.normalize factors

### PR DESCRIPTION
**Pull request by francisco-dlp**
`MVA.normalize_factors` was not rescaling properly the loadings when `on_loadings=True` (which was anyway broken, see #587).

This PR also deprecates `normalize_factors` in favor of the new `normalize_decomposition_components` and `normalize_bss_components`.

### CODE REVIEW

File: hyperspy/learn/mva.py [hunk 1]
```diff
@@ -560,13 +572,64 @@ def normalize_factors(self, which='bss', by='area', sort=True):
         else:
             raise ValueError("by must be max or mean")
 
-        factors /= by(factors, 0)
-        loadings *= by(factors, 0)
+        coeff = by(factors, 0)
+        factors /= coeff
+        loadings *= coeff
```
Comment (to266) [inline hyperspy/learn/mva.py]: Why not use the `_normalize_components`?
I guess 
```python
_normalize_components(factors, loadings, by)
```
should do

**Comment by to266** [comment]
also needs solving the merge conflicts.. otherwise looks good

File: hyperspy/learn/mva.py (hunk 1 above)
Comment (francisco-dlp) [inline hyperspy/learn/mva.py]: I prefer it in this way because it is more explicit and not much longer.

Comment (francisco-dlp) [inline hyperspy/learn/mva.py]: Sorry for replying in a rush, you're right, I'll fix that.

**Comment by francisco-dlp** [comment]
It should be fine now, thanks for the review.

**Pull request merged**